receiver/awsxrayreceiver/internal/translator/translator.go

```diff
@@ -189,3 +226,27 @@ func totalSegmentsCount(seg awsxray.Segment) int {
 
 	return 1 + subsegmentCount
 }
+
+/*
+decodeXRayTraceID decodes the traceid from xraysdk
+one example of xray format: "1-5f84c7a1-e7d1852db8c4fd35d88bf49a"
+decodeXRayTraceID transfers it to "5f84c7a1e7d1852db8c4fd35d88bf49a" and decode it from hex
+*/
+func decodeXRayTraceID(traceID *string) ([]byte, error) {
+	if traceID == nil {
+		return nil, errors.New("traceID is null")
+	}
+	if len(*traceID) < 35 {
+		return nil, errors.New("traceID length is wrong")
+	}
+	traceIDtoBeDecoded := (*traceID)[2:10] + (*traceID)[11:]
+	return hex.DecodeString(traceIDtoBeDecoded)
+}
+
+// decodeXRaySpanID decodes the spanid from xraysdk
+func decodeXRaySpanID(spanID *string) ([]byte, error) {
+	if spanID == nil {
```

Comment:
Can you check the length here too?

Reply:
@anuraaga  thx! 

is the XRay Span Id length always 16?  
i'm a little concerned that adding length limit here could potentially break span id if XRay changes the length.

Reply:
Yeah it is - OTel span ID, which we are converting to here, always has the length so X-Ray can't change it really. If something comes up in the distant future, we'll revisit then

Reply:
okay, let me add that limit then!

Reply:
Did you fix this comment? Does not seem to be a commit after.

Reply:
Will do that today

Reply:
done